### PR TITLE
[gpgerror] build on Windows using vanilla source

### PR DESCRIPTION
**Pull request by wrobelda**
This replaces usage of SMP fork with a vanilla source for Windows builds

Depends on:
 - [x] #20393 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Removes support for ARM on UWP and Windows (currently fails to build). https://dev.gnupg.org/T5687 for the reference.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

**Comment by wrobelda** [comment]
ARM cross-compilation reported upstream: https://dev.gnupg.org/T5687

**Comment by wrobelda** [comment]
Regarding the x64-uwp, the error is:

```
compile cl.exe -E -I. -I.././../src/error-1.43-dfb1333f1a.clean/src -I..   _gpg-error.def.h | \
  grep -v '^#' >gpg-error.def
rm _gpg-error.def.h
Microsoft (R) Incremental Linker Version 14.29.30137.0
Copyright (C) Microsoft Corporation.  All rights reserved.

/out:mkheader.exe 
mkheader.obj 
WindowsApp.lib WindowsApp.lib /MANIFEST /DYNAMICBASE /WINMD:NO /APPCONTAINER 
LINK : fatal error LNK1104: cannot open file 'LIBCMT.lib'
```

Looks like a Spectre-mitigation issue: https://stackoverflow.com/questions/9356135/link-fatal-error-lnk-1104-cannot-open-file-libcmt-lib#comment106230725_53212863

File: ports/libgpg-error/portfile.cmake [hunk 1]
```diff
@@ -1,97 +1,38 @@
...
+        --disable-doc
+        --disable-silent-rules
+)
+
+vcpkg_install_make()
+vcpkg_fixup_pkgconfig() 
+vcpkg_copy_pdbs()
+
+if (VCPKG_TARGET_IS_WINDOWS)
+    file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/gpg-error.lib" "${CURRENT_PACKAGES_DIR}/debug/lib/gpg-errord.lib")
```
[the remark this answers is not in the document]
Comment (wrobelda) [reply ports/libgpg-error/portfile.cmake:29]: @cenit fixed, thanks

**Comment by wrobelda** [comment]
> Regarding the x64-uwp, the error is:
> 
> ```
> compile cl.exe -E -I. -I.././../src/error-1.43-dfb1333f1a.clean/src -I..   _gpg-error.def.h | \
>   grep -v '^#' >gpg-error.def
> rm _gpg-error.def.h
> Microsoft (R) Incremental Linker Version 14.29.30137.0
> Copyright (C) Microsoft Corporation.  All rights reserved.
> 
> /out:mkheader.exe 
> mkheader.obj 
> WindowsApp.lib WindowsApp.lib /MANIFEST /DYNAMICBASE /WINMD:NO /APPCONTAINER 
> LINK : fatal error LNK1104: cannot open file 'LIBCMT.lib'
> ```
> 
> Looks like a Spectre-mitigation issue: https://stackoverflow.com/questions/9356135/link-fatal-error-lnk-1104-cannot-open-file-libcmt-lib#comment106230725_53212863

@Neumann-A would you be able to takie this over from me? I am out of free time recently and this rewrite was almost complete.

**Comment by Neumann-A** [comment]
> except ARM on UWP and Windows fails to build due to error quoted.

the error looks like the port is using `CC_FOR_BUILD` and the portfile is not correctly setup to deal with crosscompilation
`LINK : fatal error LNK1104: cannot open file 'LIBCMT.lib'` basically means something is not getting the correct build flags and has the implicit default `/DEFAULTLIB:LIBCMT.lib`

**Comment by JackBoosY** [comment]
Please temporary close this PR if you don't have spare time to finish it.

**Comment by JackBoosY** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Comment by wrobelda** [comment]
@JackBoosY can you reopen this PR, please?

**Comment by LilyWangLL** [comment]
@wrobelda We can't reopen this PR, because the GitHub remind **this branch was force-pushed or recreated** as below:
image

**Pull request closed without merging**